File: ea-domain/src/main/java/com/nbe2/domain/posts/entity/Comment.java
package com.nbe2.domain.posts.entity;

import jakarta.persistence.*;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import com.nbe2.domain.global.BaseTimeEntity;
import com.nbe2.domain.user.User;

@Entity
@Getter
@Table(name = "comments")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Comment extends BaseTimeEntity {

    @Id
    @GeneratedValue
    @Column(name = "comment_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private Post post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    private String content;

    @Builder
    private Comment(final Post post, final User user, final String content) {
        this.post = post;
        this.user = user;
        this.content = content;
    }

    public Long getPostId() {
        return post.getId();
    }

    public Long getUserId() {
        return user.getId();
    }

    public String getUserName() {
        return user.getName();
    }

    // business logic **//
    public Long update(final String content) {
        this.content = content;
        return id;
    }
}

File: ea-domain/src/main/kotlin/com/nbe3/domain/posts/Comment.kt
package com.nbe3.domain.posts

import com.nbe3.domain.global.BaseTimeEntity
import jakarta.persistence.*

@Entity
@Table(name = "comments")
class Comment private constructor(post: Post, user: User, content: String) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_id")
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    open var post: Post = post

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    open val user: User = user

    open var content: String = content

    init {
        setPost(post)
        this.user = user
        this.content = content
    }

    val postId: Long?
        get() = post!!.id

    val writerId: Long
        get() = user.getId()

    val writerName: String
        get() = user.getName()

    // ** 연관관계 편의 메서드**//
    private fun setPost(post: Post) {
        this.post = post
        post.addComment(this)
    }

    // ** business logic **//
    fun update(content: String): Long? {
        this.content = content
        return id
    }

    companion object {
        fun create(post: Post, user: User, content: String): Comment {
            return Comment(post, user, content)
        }
    }
}
